week4/routes/cohorts.js

```diff
@@ -0,0 +1,182 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../db/client") 
+
+router.get('/new', (req, res)=>{
+    res.render('new')
+})
+
+router.post('/', (req, res)=>{
+    const logoUrl = req.body.logoUrl;
+    const name = req.body.name;
+    const members = req.body.members;
+    const newCohort = {
+        logoUrl: logoUrl,
+        name: name,
+        members: members
+    };
+
+    knex 
+        .insert(newCohort)
+        .into('cohorts')
+        .returning("*")
+        .then((cohorts)=>{
+            console.log(cohorts);
+            const [cohort] = cohorts;
+            res.redirect(`/cohorts/${cohort.id}`)
+        })
+})
+
+router.get('/:id/form', (req, res)=>{
```

Comment:
since you are creating custom, non-REST routes, it would be helpful to comment your code a little more so that I can (or another reader of your code can) get a sense of what this function is meant to do before they even start, so they don't have to spend time figuring out what it is supposed to do, and then check if it does what it is meant to. Does that make sense?

Reply:
Additionally, this didn't need to be and shouldn't be a separate function from the `show` function located at `/cohorts/:id`.
You could have just extracted the get params from the query, and if they exist then you run your `teamBuilder` function, to get the teams to display them and if there were no query params, then just display the empty show page (without the teams already made up).